src/ui/MultiSelect/MultiSelect.tsx

```diff
@@ -91,16 +90,11 @@ const MultiSelect: React.FC<IMultiSelectProps> = ({
         {value.length > 0 && (
           <TagsWrapper onClick={(e) => e.stopPropagation()}>
             {value?.map((service) => (
-              <Tag key={service.key}>
-                <span>{intl(service.name)}</span>
-                <div className="icon-close-wrapper">
-                  <IoClose
-                    className="icon-close"
-                    size={20}
-                    onClick={() => handleRemoveService(service)}
-                  />
-                </div>
-              </Tag>
+              <Tag
+                key={service.key}
+                name={intl(service.name)}
+                onRemove={() => handleRemoveService(service)}
+              />
```

Comment:
Use [Event delegation](https://rahultomar092.medium.com/event-delegation-in-react-4e0083c6012e) instead of having each similar component have its own event.

Reply:
Done!!!
This was a good improvement in performance.